src/main/java/org/candlepin/controller/PoolManager.java

```diff
@@ -1541,37 +1549,31 @@ public void regenerateDirtyEntitlements(Iterable<Entitlement> entitlements) {
     }
 
     /**
-     * List entitlement pools.
+     * Lists available entitlement pools that match the provided criteria.
+     *
+     * @param key
+     *  the activation key to filter by
      *
-     * If a consumer is specified, a pass through the rules will be done for
-     * each potentially usable pool.
+     * @param includeWarnings
+     *  true to include pools with warnings, or false to exlude those pools
      *
-     * @param consumer Consumer being entitled.
-     * @param ownerId Owner whose subscriptions should be inspected.
-     * @param productId only entitlements which provide this product are included.
-     * @param activeOn Indicates to return only pools valid on this date.
-     *        Set to null for no date filtering.
-     * @param includeWarnings When filtering by consumer, include pools that
-     *        triggered a rule warning. (errors will still be excluded)
-     * @param filters builds and applies all filters when looking up pools.
-     * @param pageRequest used to determine if results paging is required.
-     * @return List of entitlement pools.
+     * @param qualifier
+     *  a qualifier with defined criteria for determining which pools should be returned
+     *
+     * @param pageRequest
+     *  containing paging information
+     *
+     * @return a list of pools that meet the defined criteria
      */
-    public Page<List<Pool>> listAvailableEntitlementPools(Consumer consumer,
-        ActivationKey key, String ownerId, String productId, String subscriptionId, Date activeOn,
-        boolean includeWarnings, PoolFilterBuilder filters,
-        PageRequest pageRequest, boolean addFuture, boolean onlyFuture, Date after) {
-
-        // Only postfilter if we have to
-        boolean postFilter = consumer != null || key != null;
+    public Page<List<Pool>> listAvailableEntitlementPools(ActivationKey key, boolean includeWarnings,
```

Comment:
I see what you are saying, and I think we can add the rest of the parameters into the `PoolQualifier` object. 

Also, I wasn't aware of `QueryArguments`, but it might make sense to have the `PoolQualifier` object extend that class and I think we can get rid of using the `PageRequest` object! I can work on making those changes.

Reply:
Added the additional arguments into the `PoolQualifier` and removed the use of `PageRequest` by having the `PoolQualifier` extend `QueryArguments`